### PR TITLE
Names+extras

### PR DESCRIPTION
**Pull request by ColonelThirtyTwo**
Store object names and extra data from gltf to the easy structs.

### CODE REVIEW

File: src/scene/camera.rs [hunk 1]
```diff
@@ -4,6 +4,14 @@ use gltf::camera::Projection;
 /// Contains camera properties.
 #[derive(Clone, Debug)]
 pub struct Camera {
+    #[cfg(feature="names")]
+    /// Camera name. Requires the `names` feature.
+    pub name: Option<String>,
+
+    #[cfg(feature="extras")]
+    /// Scene extra data. Requires the `extras` feature.
```
Comment (ColonelThirtyTwo) [inline src/scene/camera.rs:12]: Typo, should be "camera extra data"

File: src/scene/model/mod.rs [hunk 2]
```diff
@@ -237,10 +272,17 @@ impl Model {
...
+        mesh: &gltf::Mesh,
+        primitive_index: usize,
         primitive: gltf::Primitive,
         transform: &Matrix4<f32>,
         data: &mut GltfData,
     ) -> Self {
+        #[cfg(not(feature="names"))]
+        {
+            let _ = mesh;
+        }
```
Comment (flomonster) [inline src/scene/model/mod.rs:284]: Instead, rename `mesh` argument to `_mesh` this will disable warnings if "names" feature isn't used.

**Review by flomonster — CHANGES_REQUESTED** [review]

File: src/scene/model/mod.rs (hunk 2 above)
Comment (ColonelThirtyTwo) [reply src/scene/model/mod.rs:284]: The underscore silences other errors though. Doing an unused assignment in a condition silences the unused variable warning only in the case where `names` is off, while keeping every other warning active (including the unused variable warning when `names` is on and the variable should be used.

**Comment by flomonster** [comment]
Merged! Thanks for your contribution.

**Pull request closed without merging**